arviz/data/base.py

```diff
@@ -20,7 +26,8 @@ def __call__(self, func):  # noqa: D202
         def wrapped(cls, *args, **kwargs):
```

Comment:
Add `functools.wrap`?

Reply:
I only modified slightly the behaviour of `requres` class and tried to modify it as little as possible. I can definitely change it to use `functools.wrap` unless @aloctavodia, @ColCarroll, @ahartikainen, @canyon289 have some reason against it. 

For example, in `az.interactive_backend`, `contextlib` cannot be used as it does not work properly with matplotlib backend changes, thus, I followed [`mpl.rc_context`](https://matplotlib.org/3.1.1/_modules/matplotlib.html#rc_context). I can't tell if there is some reason that I cannot see behind `requires` implementation.